frontend/src/App.tsx

```diff
@@ -1,19 +1,47 @@
 import "./App.css";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 
 import Input from "./components/Input";
 import TodoList from "./components/TodoList";
 import { Todo } from "./models/todo.model";
 
+const urlString = "http://localhost:3001/todos/";
+
 function App() {
   const [todos, setTodos] = useState<Todo[]>([]);
   const [newTodoDescription, setNewTodoDescription] = useState("");
+  const [loading, setLoading] = useState(true);
 
-  const onChange = (name: string, value: string) => {
-    setNewTodoDescription(value);
-  };
+  function fetchTodos(url: string, signal: AbortSignal) {
+    return fetch(url, {
+      signal,
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+  }
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const loadTodos = async () => {
+      try {
+        const response = await fetchTodos(urlString, controller.signal);
+        const tempTodos: Todo[] = await response.json();
+        setTodos(tempTodos);
+        setLoading(false);
```

Comment:
There is a problem here. If the request fails for some reason, `loading` will not be set to `false`, and the user will always see the loading state.

Fix this such that `loading` is set to `false` in both the cases.